apps/emqx_cluster_link/src/emqx_cluster_link.erl

```diff
@@ -94,13 +98,20 @@ on_message_publish(
         {route_updates, #{actor := Actor}, RouteOps} ->
             ok = update_routes(ClusterName, Actor, RouteOps);
         {heartbeat, #{actor := Actor}} ->
-            ok = actor_heartbeat(ClusterName, Actor)
+            ok = actor_heartbeat(ClusterName, Actor);
+        {unlink, InfoMap} ->
+            _ = on_unlinking(ClusterName, InfoMap);
```

Comment:
I'd definitely argue against having this on the protocol level, at least in the first iteration.

This looks like an optimization that moderately (and shortly) impacts performance, but greatly increases operational model complexity, and thus the chance a cluster ends up in a half-broken state: imagine all the different interleavings of ensure / remove RPCs hitting different nodes in the cluster when a user quickly hits _disable_ and _enable_ button quickly one after another on the remote cluster. I'd avoid having that, and all the logic, corner cases and associated "shadow configuration" BPAPIs that lack transactional semantics.

Reply:
Removed.